lib/fields/class-wp-rest-meta-fields.php

```diff
@@ -222,7 +222,19 @@ protected function update_meta_value( $object, $name, $value ) {
 			);
 		}
 
-		if ( ! update_metadata( $this->get_meta_type(), $object, wp_slash( $name ), wp_slash( $value ) ) ) {
+		$meta_type  = $this->get_meta_type();
+		$meta_key   = wp_slash( $name );
+		$meta_value = wp_slash( $value );
+
+		// Do the exact same check for a duplicate value as in update_metadata() to avoid update_metadata() returning false.
+		$old_value = get_metadata( $meta_type, $object, $meta_key );
+		if ( 1 === count( $old_value ) ) {
```

Comment:
@joehoyle No, `update_metadata()` only returns `false` if you are updating a single meta value and the previous and new values are the same.